### PR TITLE
[#48] More timeouts and dependencies for k8s_setup step

### LINKED ISSUE #48: Add more timeouts
We should add some timeouts on `k8s_setup` step, especially on `helm_release` resources (for example - `istio`, `monitoring`).
Sometimes 5 minutes by default is not enough.

### PR DESCRIPTION
**Pull request by maxkochubey**
This PR fixes #48.

### CODE REVIEW

File: pipelines/gcp-create-cluster.Jenkinsfile [hunk 1]
```diff
@@ -42,20 +42,19 @@ pipeline {
                     sshagent(["${env.legionProfilesGitlabKey}"]) {
                         sh"""#!/bin/bash -ex
                         #TODO get repo url from passed parameters
-                        mkdir -p \$(getent passwd \$(whoami) | cut -d: -f6)/.ssh && ssh-keyscan git.epam.com >> \$(getent passwd \$(whoami) | cut -d: -f6)/.ssh/known_hosts
+                        #mkdir -p \$(getent passwd \$(whoami) | cut -d: -f6)/.ssh && ssh-keyscan git.epam.com >> \$(getent passwd \$(whoami) | cut -d: -f6)/.ssh/known_hosts
```
Comment (mcdoker18) [inline pipelines/gcp-create-cluster.Jenkinsfile]: Remove the command if you don't need it

File: pipelines/gcp-create-cluster.Jenkinsfile [hunk 2]
```diff
@@ -42,20 +42,19 @@ pipeline {
...
                         fi
                         cd legion-profiles && git checkout ${env.param_legion_profiles_branch}
                         """
                     }
                     // Checkout CICD repo with private DNS zone
                     sshagent(["${env.legionCicdGitlabKey}"]) {
                         sh"""
-
-                        mkdir -p \$(getent passwd \$(whoami) | cut -d: -f6)/.ssh && ssh-keyscan git.epam.com >> \$(getent passwd \$(whoami) | cut -d: -f6)/.ssh/known_hosts
+                        #mkdir -p \$(getent passwd \$(whoami) | cut -d: -f6)/.ssh && ssh-keyscan git.epam.com >> \$(getent passwd \$(whoami) | cut -d: -f6)/.ssh/known_hosts
```
Comment (mcdoker18) [inline pipelines/gcp-create-cluster.Jenkinsfile]: Remove the command if you don't need it

File: terraform/modules/k8s/istio/main.tf [hunk 3]
```diff
@@ -32,11 +32,12 @@ resource "helm_release" "istio-init" {
   version    = var.istio_version
   namespace  = var.istio_namespace
   repository = "istio"
+  depends_on = [kubernetes_namespace.istio]
 }
 
 resource "null_resource" "delay" {
   provisioner "local-exec" {
-    command = "timeout 200 bash -c 'until $(kubectl get --all-namespaces gateways.networking.istio.io && kubectl get --all-namespaces envoyfilters.networking.istio.io && kubectl get --all-namespaces policies.authentication.istio.io && kubectl get --all-namespaces destinationrules.networking.istio.io && kubectl get --all-namespaces virtualservices.networking.istio.io && kubectl get --all-namespaces envoyfilters.networking.istio.io && kubectl get --all-namespaces attributemanifests.config.istio.io && kubectl get --all-namespaces handlers.config.istio.io && kubectl get --all-namespaces meshpolicies.authentication.istio.io); do sleep 5; done'"
+    command = "timeout 200 bash -c 'until [ $(kubectl get customresourcedefinitions -l release=istio --no-headers | wc -l) -ge 23 ]; do sleep 5; done'"
```
Comment (mcdoker18) [inline terraform/modules/k8s/istio/main.tf]: Please, extract the `23` value to a local variable

**Review by mcdoker18 — CHANGES_REQUESTED** [review]

**Review by karbyshevds — APPROVED** [review]

File: pipelines/gcp-create-cluster.Jenkinsfile (hunk 1 above)
Comment (maxkochubey) [reply pipelines/gcp-create-cluster.Jenkinsfile]: Done in b95ec91.

File: pipelines/gcp-create-cluster.Jenkinsfile (hunk 2 above)
Comment (maxkochubey) [reply pipelines/gcp-create-cluster.Jenkinsfile]: Done in b95ec91.

File: terraform/modules/k8s/istio/main.tf (hunk 3 above)
Comment (maxkochubey) [reply terraform/modules/k8s/istio/main.tf]: Done in 82b149c.

**Review by mcdoker18 — APPROVED** [review]

**Pull request merged**